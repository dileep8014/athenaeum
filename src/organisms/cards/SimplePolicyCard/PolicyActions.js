import React from 'react';
import PropTypes from 'prop-types';
import Layout from 'atoms/Layout';
import Button from 'atoms/Button';
import styles from './policy_card.module.scss';

const VIEW_POLICY_TEXT = 'View Policy';

export const PolicyActions = (props) => {
  const { onDetails } = props;

  return (
    <div className={styles['actions']}>
      <Layout smallCols={[ 12 ]} style={{ width: '100%' }}>
        <Button
          onClick={onDetails}
        >
          {VIEW_POLICY_TEXT}
        </Button>
      </Layout>
    </div>
  );
};

PolicyActions.propTypes = {
  onDetails: PropTypes.func
};

export default PolicyActions;
