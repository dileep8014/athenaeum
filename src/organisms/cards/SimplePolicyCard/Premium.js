import React from 'react';
import PropTypes from 'prop-types';
import accounting from 'accounting';
import Text from 'atoms/Text';
import styles from './policy_card.module.scss';

export const Premium = ({ premium }) => {

  const formattedPremium = accounting.formatMoney(premium.price);

  return (
    <div className={styles['premium']}>
      {
        premium.price ? (
          <Text
            size={4}
            font='a'
            className={styles['premium-text']}
          >
            {formattedPremium}
            {' '}
            <Text tag='span' size={11} font='a' spaced color='neutral-2'>{`/${premium.format.toUpperCase()}`}</Text>
          </Text>
        )
          :
          <Text size={7} font='a'>{premium.defaultText}</Text>
      }

      {premium.tooltip}
    </div>
  );
};

Premium.propTypes = {
  premium: PropTypes.object
};

export default Premium;
