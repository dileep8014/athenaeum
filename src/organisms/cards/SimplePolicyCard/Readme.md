### SimplePolicyCard Example:

```jsx
  <div>
    <SimplePolicyCard
      carrierLogo={
        <img
          src='https://cdn-staging.policygenius.com/assets/insurance-cards-logos/banner-life-dark-2x-c02aef5dc27115997f3185217e684806.png'
          alt='hello'
        />
      }
      premium={{
        price: 13,
        format: 'mo',
        defaultText: 'Quote available from a PolicyGenius expert'
      }}
      onContinue={() => alert('continue button clicked')}
      onDetails={() => alert('details button clicked')}
      compareCheckbox={{
        onCompare: (e) => setState({ compare: e.target.checked }),
        compareSelected: state.compare,
        name: 'unique-name'
      }}
    />

    <Spacer size={24} />

    <SimplePolicyCard
      carrierLogo={
        <img
          src='https://cdn-staging.policygenius.com/assets/insurance-cards-logos/banner-life-dark-2x-c02aef5dc27115997f3185217e684806.png'
          alt='hello'
        />
      }
      premium={{
        price: 13,
        format: 'mo',
        defaultText: 'Quote available from a PolicyGenius expert'
      }}
      onContinue={() => alert('continue button clicked')}
      onDetails={() => alert('details button clicked')}
      compareCheckbox={{
        onCompare: (e) => setState({ compare: e.target.checked }),
        compareSelected: true,
        name: 'unique-name-2'
      }}
    />
  </div>
```
