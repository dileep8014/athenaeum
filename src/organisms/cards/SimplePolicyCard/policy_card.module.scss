$border: #dfdfdf;
$footer: #f6f6f6;

@mixin section($grow, $basis, $centered) {
  flex: #{$grow} 1 #{$basis};
  padding: 0 ru(1.25);
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: relative;

  @if $centered {
    align-items: center;
  }
}

.policy-card {
  border: 1px solid $border;
  background-color: color('neutral-8');
  display: none;

  &.selected {
    border: 1px solid color('secondary-2');
  }
}

.body {
  padding: rem-calc(20px) rem-calc(36px) ru(1) rem-calc(12px);
  display: flex;
  align-items: center;
}

.divider {
  height: ru(3.25);
  border-left: 1px solid $border;
}

.node {
  display: inline-block;
  @include typography(7);
  color: color('neutral-3');
  padding-right: ru(.5);
}

.checkbox-wrapper {
  display: flex;
  position: relative;
  flex-direction: column;
  align-items: center;
  padding: rem-calc(8px) 0;

  &:after {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    background: transparent;
    content: '';
    cursor: pointer;
  }
}

.checked {
  font-weight: 600;
}

.carrier-img {
  align-items: center;
  display: flex;
  height: rem-calc(42px);
  justify-content: center;
  max-width: rem-calc(127px);

  img {
    max-width: 100%;
    max-height: 100%;
  }
}

.premium {
  text-align: center;
  white-space: nowrap;
}

.policy-tooltip {
  > * {
    display: flex;
    align-items: center;
  }
}

.policy-type-hover-message {
  left: 50%;
  top: calc(100% + 18px);
  width: ru(11.5);
}

.policy-type-tooltip {
  cursor: help;
}

/* IE 11 hack to make sure svg does not take up 100% height */
.svg-text {
  max-height: ru(1);
}

.actions p {
  font-size: 15px;
  white-space: nowrap;
}

@media #{$small-only} {
  .policy-tooltip {
    > * {
      justify-content: center;
    }
  }

  .body {
    flex-direction: column;
  }

  .footer {
    justify-content: center;
    text-align: center;
  }

  .actions {
    padding: ru(1);
    text-align: center;
    border-top: 1px solid $border;
    width: 100%;
  }

  .policy-type {
    padding: ru(1);
    text-align: center;
  }

  .carrier-logo {
    text-align: center;
  }

  .policy-name {
    max-width: 140px;
    margin: auto;
    margin-bottom: ru(.5);
    line-height: ru(1);
  }

  .node {
    &:first-child {
      padding-right: ru(.25);
    }
  }
}

@media #{$small-and-medium-only} {
  .continue {
    margin-bottom: ru(.75);
  }
}

@media #{$medium-up} {
  .policy-card {
    display: block;
  }

  .mobile-card {
    display: none;
  }

  .compare {
    @include section(0, 17.5%, true);
    label { margin-left: rem-calc(8px); }
    padding: 0;
  }

  .carrier-logo {
    @include section(1, 20%, true);
    padding: 0 rem-calc(12px);
  }

  .premium {
    @include section(1, 25%, false);
    padding: 0 rem-calc(12px);
  }

  .actions {
    @include section(1, 25%, true);
    text-align: center;
    padding: 0 0 0 rem-calc(6px);
  }
}

@media #{$large-up} {
  .compare {
    max-width: ru(5.25);
  }

  .carrier-logo {
    padding: 0px rem-calc(18px);
  }

  .info-divider {
    display: block;
  }
}
