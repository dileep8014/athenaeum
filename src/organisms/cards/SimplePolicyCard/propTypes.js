import PropTypes from 'prop-types';

export default {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  /**
   * The insurance carrier logo node
   */
  carrierLogo: PropTypes.node.isRequired,

  /**
   * The total number of customers for the carrier
   */
  totalCustomers: PropTypes.shape({
    value: PropTypes.number.isRequired,
    hoverMessage: PropTypes.node.isRequired
  }),
  /**
   * The insurance premium
   */
  premium: PropTypes.shape({
    price: PropTypes.number.isRequired,
    format: PropTypes.string.isRequired,
    defaultText: PropTypes.node,
  }),
  /**
   * Optional discount percentage
   */
  discount: PropTypes.node,
  /**
   * Optional array of nodes that will be spaced appropriately in the footer
   */
  footer: PropTypes.array,
  /**
   * Optional array of nodes that will be spaced appropriately in the header
   *
   * Only displayed on mobile
   */
  header: PropTypes.array,
  /**
   * Click handler for details button
   */
  onDetails: PropTypes.func,
  /**
   * Props for the Checkbox to compare
   *
   * NOTE: `name` must be unique
   */
  compareCheckbox: PropTypes.shape({
    onCompare: PropTypes.func.isRequired,
    compareSelected: PropTypes.bool,
    name: PropTypes.string,
  }),

  /**
   * Boolean to toggle styling for Continue CTA if any other card in the group is selected
   */
  anyCardSelected: PropTypes.bool,
};
