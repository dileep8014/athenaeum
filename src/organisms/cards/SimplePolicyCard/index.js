import React from 'react';
import classnames from 'classnames';

import SimpleFeaturedPolicyCard from 'organisms/cards/SimpleFeaturedPolicyCard';

import styles from './policy_card.module.scss';
import propTypes from './propTypes';
import Compare from './Compare';
import CarrierLogo from './CarrierLogo';
import Premium from './Premium';
import PolicyActions from './PolicyActions';

function SimplePolicyCard(props) {
  const {
    carrierLogo,
    premium,
    onContinue,
    compareCheckbox,
    className,
    anyCardSelected,
  } = props;

  const policyCardClasses = classnames(
    styles['policy-card'],
    compareCheckbox.compareSelected && styles['selected'],
    className
  );

  return (
    <div>
      <div className={styles['mobile-card']}>
        <SimpleFeaturedPolicyCard
          premium={premium}
          carrierLogo={carrierLogo}
          onContinue={onContinue}
          compareCheckbox={compareCheckbox}
        />
      </div>
      <div className={policyCardClasses}>
        <div className={styles['body']}>
          <Compare {...compareCheckbox} />
          <div className={styles['divider']} />
          <CarrierLogo carrierLogo={carrierLogo} />
          <div className={styles['divider']} />
          <Premium premium={premium} />
          <PolicyActions
            onContinue={onContinue}
            onCompare={compareCheckbox.onCompare}
            premium={premium}
            selected={compareCheckbox.compareSelected || anyCardSelected}
          />
        </div>
      </div>
    </div>
  );
}

SimplePolicyCard.propTypes = propTypes;

export default SimplePolicyCard;
