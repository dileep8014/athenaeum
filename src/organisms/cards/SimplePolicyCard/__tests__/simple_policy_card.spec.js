import React from 'react';
import renderer from 'react-test-renderer';

import SimplePolicyCard from 'organisms/cards/SimplePolicyCard';

describe('<SimplePolicyCard />', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      carrierLogo: <div></div>,
      premium: {
        price: 19.8,
        format: 'mo'
      },
      onDetails: jest.fn(),
      compareCheckbox: {
        onCompare: jest.fn()
      }
    };
  });

  it('renders correctly', () => {
    const actual = renderer.create(<SimplePolicyCard {...defaultProps} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
