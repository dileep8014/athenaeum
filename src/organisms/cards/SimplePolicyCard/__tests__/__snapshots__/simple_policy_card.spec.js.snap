// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimplePolicyCard /> renders correctly 1`] = `
<div>
  <div
    className="mobile-card"
  >
    <div
      className="featured-policy-card"
    >
      <div
        className="content"
      >
        <div
          className="premium"
        >
          <p
            className="primary-3 type-a-3-bold inherit"
          >
            $19.80
             
            <strong
              className="bold neutral-2 type-b-7-medium inherit"
            >
              /MO
            </strong>
          </p>
        </div>
        <div
          className="top-6"
        />
        <div
          className="carrier-logo"
        >
          <div />
        </div>
        <div
          className="top-6"
        />
        <div
          className="button-group"
        >
          <button
            className="button"
            disabled={undefined}
            onClick={[Function]}
            type="button"
          >
            View Policy
          </button>
          <div
            className="top-2"
          />
          <button
            className="button outline"
            disabled={undefined}
            onClick={[Function]}
            type="button"
          >
            Compare
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="policy-card"
  >
    <div
      className="body"
    >
      <div
        className="hide compare small"
      >
        <div
          className="checkbox-wrapper"
          id="undefined-checkbox-wrapper"
          onClick={[Function]}
        >
          <p
            className="primary-3 type-b-10-medium inherit"
          >
            Compare
          </p>
          <div
            className="top-1"
          />
          <label
            className="checkbox"
            htmlFor="checkbox-undefined"
          >
            <input
              checked={undefined}
              className="checkbox-input"
              id="checkbox-undefined"
              name={undefined}
              type="checkbox"
              value={undefined}
            />
            <span
              className="checkbox-label"
            />
          </label>
        </div>
      </div>
      <div
        className="divider"
      />
      <div
        className="carrier-logo"
      >
        <div />
      </div>
      <div
        className="divider"
      />
      <div
        className="premium"
      >
        <p
          className="primary-3 premium-text type-a-4-bold inherit"
        >
          $19.80
           
          <strong
            className="bold neutral-2 type-b-11-medium inherit"
          >
            /MO
          </strong>
        </p>
      </div>
      <div
        className="actions"
      >
        <div
          className="layout"
          id={undefined}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div
            className="col col-sm-12 border-neutral-6"
            onClick={undefined}
            style={undefined}
          >
            <button
              className="button"
              disabled={undefined}
              onClick={[Function]}
              type="button"
            >
              View Policy
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
